data_dictionary/storage_options.hh

```diff
@@ -32,7 +32,15 @@ struct storage_options {
         std::map<sstring, sstring> to_map() const;
         bool operator==(const s3&) const = default;
     };
-    using value_type = std::variant<local, s3>;
+    struct mirrored {
+        sstring bucket;
+        sstring endpoint;
+        static constexpr std::string_view name = "MIRRORED";
```

Comment:
In other words -- we need more flexibility in describing which component to cache rather than just "this keyspace is MIRRORED"

Reply:
Another consideration here -- there's an option for partial caching. It's questionable whether we're going to implement it at all, but still -- partial/full caching is also not an option for the keyspace, but for a storage/components

Reply:
i did have caching mode in this PR, but it was dropped because of the discussion we had yesterday. could you please help me understand what on earth we are after? a mirroring storage or multiple caching mode, and at which granularity ?

Reply:
There are two layers here. First -- S3 storage option (from create keyspace query). Second -- S3 storage type (the child class of class storage). I'm fine with the implementation of yet another storage class, no objections here. I'm even fine if we throw away s3 storage class and make mirrored storage class be the one backing S3 storage option.

My point here is that the "mirroring" thing should have more flexibility rather than on/off, because we don't have one single requirement on what and how to mirror.

For example, let's assume we merge this PR as is. It implements mirroring for all sstables components once keyspace is created with storage options, type = "MIRRORED". The next step might be to implement mirroring for Index and Filter files only and keep other components as pure S3 files. What do we do then, implement "MIRRORED_LITE" type in storage options? Or add more stuff to storage options itself? If that's the case, what's going to be the difference between "storage options {type = S3}" vs "storage options { type= mirrored, cache=none}"?